crates/sui-indexer-alt-framework/src/ingestion/client.rs

```diff
@@ -214,3 +214,183 @@ impl IngestionClient {
         Ok(Arc::new(data))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use prometheus::Registry;
+    use std::collections::HashMap;
+    use std::sync::Arc;
+    use std::time::Duration;
+    use tokio::sync::Mutex;
+    use tokio_util::bytes::Bytes;
+
+    use crate::ingestion::test_utils::test_checkpoint_data;
+
+    use super::*;
+
+    /// Mock implementation of IngestionClientTrait for testing
+    #[derive(Default)]
+    struct MockIngestionClient {
+        checkpoints: Arc<Mutex<HashMap<u64, FetchData>>>,
+        transient_failures: Arc<Mutex<HashMap<u64, usize>>>,
+        not_found_failures: Arc<Mutex<HashMap<u64, usize>>>,
+    }
```

Comment:
Check out [`dashmap`](https://docs.rs/dashmap/latest/dashmap/index.html) -- it's a concurrent hash map implementation that you can use instead of `Mutex<HashMap<...>>` and not have to worry about locking etc.

I also notice that you are not `clone()`-ing the `MockIngestionClient` -- so do you need the `Arc`?